### PR TITLE
Add better help docs to kup

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
The help command has been extended so that each sub-command has a dedicated help doc, e.g:

image

### CODE REVIEW

**Comment by ehildenb** [comment]
This is the script that gets `curl`ed onto users systems: https://github.com/runtimeverification/k/blob/master/package/nix/install

I'm pretty sure we just need to update that script to source `kup` from this repo the first time around? Though I can't really see how to do so from that script, because I don't see where it actually downloads and install `k` or `kup` :laughing: .

But that should be the only adjustment needed, AFAICT.

**Review by ehildenb — APPROVED** [review]

File: package/nix/install [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  GREEN='\033[0;32m'
+  RED='\033[0;31m'
+  NC='\033[0m'
+
+  NIX_POST_INSTALL_MESSAGE="To ensure that the necessary environment variables are set, either log in again, or type
+
+  . $HOME/.nix-profile/etc/profile.d/nix.sh
+
+in your shell."
+
```
Comment (ehildenb) [inline package/nix/install:15]: I don't htink we need to store this script here, instead we store it in K and point it at this one to get `kup`. That's easier build-system wise, and this script shouldn't change so often anyway.

**Comment by ehildenb** [comment]
The feature-set looks good to me, except the `install` script not needing to live here. Love the detailed documentation :)

**Review by tothtamas28 — APPROVED** [review]
LGTM, great documentation. :sparkles:

**Pull request merged**